src/python/WMCore/WebTools/RESTFormatter.py

```diff
@@ -26,7 +28,21 @@ def __init__(self, config):
 
         TemplatedPage.__init__(self, config)
 
+    def genstreamer(self, data):
+        yield "["
+        firstItem  = True
+        for item in data:
+            if not firstItem:
+                yield ","
```

Comment:
Yes, this will basically create string representation of list of dicts, e.g.
[{dict1},{dict2}]

This is what DBS returns and in fact this is what
src/python/WMCore/Database/DBFormatter.py is doing in formatDict which I intend
to convert into generator as well. But first we need to test how RESTModel
class will handle generator response from db layer and unwind it into
string representation of list of dicts for a client.

On  0, Eric Vaandering <notifications@github.com> wrote:
> > @@ -26,7 +28,21 @@ def __init__(self, config):
> >  
> >          TemplatedPage.__init__(self, config)
> >  
> > +    def genstreamer(self, data):
> > +        yield "["
> > +        firstItem  = True
> > +        for item in data:
> > +            if not firstItem:
> > +                yield ","
> 
> Ah, wait, you are also guaranteed to yield the actual value, right? So no problem
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6221/files#r39165050